### ISSUE TITLE
touchAfter not working in express application

### ISSUE THREAD

**Issue by josevillacreses** [jdesboeufs/connect-mongo#199]
First of all, thank you for this project, your time and your kind help.

Abstract of my session configuration:

```
cookie: {
    maxAge: 60 * 1000 //one minute
  },
resave: false,
saveUninitialized: false,
store: new MongoStore({
    url:xxxx,
    touchAfter: 20, //20 seconds
    collection: 'sessions'
  }
```

At login, I save some data in the session.
Then I make some manual requests periodically (every 5 seconds aproximately).
The requests get some data from the session.
#### The problem: After a minute (the original maxAge) I get my data deleted in the application.

Monitoring the mongo document for the session, it gets updated as specified (approximately every 20 seconds).
At the beginning:

```
    "expires" : ISODate("2015-10-07T17:36:25.383Z"),
    "lastModified" : ISODate("2015-10-07T17:35:25.385Z")
```

Then eventually to:

```
    "expires" : ISODate("2015-10-07T17:36:48.643Z"),
    "lastModified" : ISODate("2015-10-07T17:35:48.644Z")
```

And then to:

```
    "expires" : ISODate("2015-10-07T17:37:11.732Z"),
    "lastModified" : ISODate("2015-10-07T17:36:11.733Z")
```

Printing req.session.cookie.expires and req.session.cookie.maxAge in the express application when attending the requests I get:

```
req.session.cookie.expires Wed Oct 07 2015 12:36:25 GMT-0500 (ECT)
req.session.cookie.maxAge 59358
```

At some other point in the test:

```
req.session.cookie.expires Wed Oct 07 2015 12:36:25 GMT-0500 (ECT)
req.session.cookie.maxAge 13653
```

And at another:

```
req.session.cookie.expires Wed Oct 07 2015 12:36:25 GMT-0500 (ECT)
req.session.cookie.maxAge 3887
```

Every time `req.session.cookie.expires` has the same value. I guess it should be updated.
Every time `req.session.cookie.maxAge` keeps dropping. I guess it should be updated.
##### Summing up. The session document gets updated as expected. But the req.session object in express does not act accordingly.

At least that is what I think is happening. Maybe I am configuring something wrong or I am not understanding something.

I am using:
connect-mongo version 0.8.1.
express-session version 1.11.1.
express version 4.12.3.

**Comment by josevillacreses** [comment]
In contrast, this is what I get when, in the route for the requests I invoke periodically, I include some update in the req.session object:

```
req.session.cookie.expires Wed Oct 07 2015 13:25:34 GMT-0500 (ECT)
req.session.cookie.maxAge 55784
```

```
req.session.cookie.expires Wed Oct 07 2015 13:25:34 GMT-0500 (ECT)
req.session.cookie.maxAge 55775
```

```
req.session.cookie.expires Wed Oct 07 2015 13:25:38 GMT-0500 (ECT)
req.session.cookie.maxAge 56646
```

```
req.session.cookie.expires Wed Oct 07 2015 13:25:42 GMT-0500 (ECT)
req.session.cookie.maxAge 54922
```

```
req.session.cookie.expires Wed Oct 07 2015 13:25:42 GMT-0500 (ECT)
req.session.cookie.maxAge 54917
```

As you can see, `expires` and `maxAge` get updated.

I think this should happen every 20 seconds, as specified in `touchAfter `.
Again, in the database it seems to work as expected. The problem is in the express/node application.

**Comment by ethelius** [comment]
Can corroborate when using the same session configuration with 

connect-mongo 0.8.2
express-session 1.11.3
express 4.13.3

The mongo document:
{
    "_id" : [...],
    "session" : "{\"cookie\":{\"originalMaxAge\":60000,\"expires\":\"2015-10-29T07:38:36.394Z\",\"httpOnly\":true,\"path\":\"/\"},\"passport\":{\"user\":\"55a5e4884073204d9a0541cb\"}}",
    "expires" : ISODate("2015-10-29T07:39:04.484Z"),
    "lastModified" : ISODate("2015-10-29T07:38:04.484Z")
}

From my understanding, the "expires" in the serialized session should match the expires in the document, but session.expires never gets updated when the main expires does.

I was able to get the rolling window working by jamming "updateFields.session = this.options.serialize(session);" into the MongoStore.prototype.touch function.

Not sure if it's a connect-mongo issue or if I'm using it incredibly wrong.
